app/controllers/application_controller.rb

```diff
@@ -61,15 +61,28 @@ def valid_help_actions
 
   def confirm_two_factor_setup
     user = request.env["warden"].user
-    return unless user &&
-      user.need_two_factor_authentication?(request) &&
-      !user.totp_enabled?
 
     if Rails.configuration.enable_enhanced_2fa_experience
+      return unless user
```

Comment:
`return unless user.present?` is easier to read

Reply:
or ``` if user.nil?```